### DISCUSSION TITLE
Understanding "Top 5 Volumes by IOPS" on Volume Dashboard

### DISCUSSION THREAD

**Discussion by tsohst** [NetApp/harvest#2246] [Q&A] [answered]
* We're unable to understand the values for "Top 5 Volumes by IOPS"
* Looking at the graph, we have 
  * 500 "Top 5 Volumes by IOPS"
  * 0 IOPS Read
  * 50 IOPs Write
  * 160 IOPs Other

Should Write/Read/Other compared not be the same value as Top 5 Volumes by IOPS

![image](https://github.com/NetApp/harvest/assets/32710648/95768aaa-7ee3-43db-9df9-c5b57e792028)

**Comment by rahulguptajss** [comment]
@tsohst The "Top 5" function will determine the top 5 volumes with the highest peaks in the given time range and present their corresponding time series data. Write, Read, and Other IOPS components are separate metrics, which means that their respective top 5 values may not be the same.

**Reply by tsohst** [reply]
@rahulguptajss But I've filtered the dashboard on the Top, down to a single Volume so that the dashboards and graphs should only represent values from one Volume.

**Reply by rahulguptajss** [reply]
@tsohst I think that you're referring to a UX issue. Top 5 display is determined by the TopResources variable in the Grafana dashboard, which remains unchanged even when only one volume is selected from the dropdown menu. Could you please create an issue for this?

**Reply by tsohst** [reply]
Even without the "topk" function in the graph, all 3 values combined to not match the "total ops" value.
![image](https://github.com/NetApp/harvest/assets/32710648/b7b18157-ba95-40a1-b53a-b2a381b0a824)

**Reply by rahulguptajss** [reply]
@tsohst Could you confirm your Harvest version? I have tried in my system as below and values seems to match in prometheus. Also if you can share poller logs to us @ ng-harvest-files@netapp.com

`volume_total_ops{volume="osc_iscsi_vol01"} - (volume_read_ops{volume="osc_iscsi_vol01"}+volume_write_ops{volume="osc_iscsi_vol01"}+volume_other_ops{volume="osc_iscsi_vol01"})`

![image](https://github.com/NetApp/harvest/assets/25551691/360b77b4-8dd6-455e-a216-ad8438ff2d3e)

**Reply by tsohst** [reply]
This matches the "missing" IOPs, we're running on harvest 23.05.0.
I will send over logs.
![image](https://github.com/NetApp/harvest/assets/32710648/75419e9f-fcbe-41b5-a7a8-26ff4720b123)

**Reply by rahulguptajss** [reply]
@tsohst Do you see these differences across many volumes? How many records below query returns? 

`(volume_total_ops - (volume_read_ops+volume_write_ops+volume_other_ops)) > 5`

**Reply by tsohst** [reply]
![image](https://github.com/NetApp/harvest/assets/32710648/92e70608-f0c6-4e35-b034-447d83660ff8)

**Reply by rahulguptajss** [reply]
Thanks ,It seems that there are ~6-7 volumes having big deviation. We'll check logs to see if there are any issues there.

**Reply by tsohst** [reply]
Logs are send over to you 👍

**Reply by rahulguptajss** [reply]
Thanks received.

**Reply by rahulguptajss** [reply, marked as answer]
@tsohst As discussed over email, Refer [here](https://kb.netapp.com/onprem/ontap/Performance/Why_does_the_statistics_volume_show_command_show_the_total_ops_which_donot_match_the_read%2C_write%2C_and_other_ops) for the details.
